src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -353,3 +352,26 @@ class LiteralCoercionSegment(BaseSegment):
         OneOf("DATE", "DATETIME", "TIME", "TIMESTAMP"),
         Ref("QuotedLiteralSegment"),
     )
+
+
+@bigquery_dialect.segment()
+class HyphenatedObjectReferenceSegment(ansi_dialect.get_segment("ObjectReferenceSegment")):  # type: ignore
+    type = "hyphenated_object_reference"
+    match_grammar = ansi_dialect.get_segment(
+        "ObjectReferenceSegment"
+    ).match_grammar.copy()
+    match_grammar.delimiter = OneOf(
```

Comment:
This is a bit of a hack, but it avoids having to copy/paste lots of code from ANSI ObjectReferenceSegment. Should we consider having the `copy()` function support this directly somehow?

Reply:
Very interesting question. I'm not aware of doing this a better way 🤔 

I'm ok with this as it is but I agree it feels a little gross.